tests/test_main.py

```diff
@@ -113,3 +114,12 @@ def test_run_match_config_params() -> None:
         if key not in ("app_dir",)
     }
     assert config_params == run_params
+
+
+@pytest.mark.anyio
+async def test_run_invalid_host(caplog: pytest.LogCaptureFixture) -> None:
+    with pytest.raises(SystemExit):
+        config = Config(app=app, host="illegal_host")
+        server = Server(config=config)
+        await server.serve()
+    assert "ERROR" == caplog.records[-1].levelname
```

Comment:
Reviewer,

The error message is different on different OSes. I asserted the log level instead, is this okay?

Reply:
Yes. That is the purpose.

Reply:
```suggestion
```

Reply:
Can you also add an assertion on the `code` from the `SystemExit`? :pray:

e.g.

```python
assert exc.code == 1
```